doc/sphinx/proofs/writing-proofs/reasoning-inductives.rst

```diff
@@ -5,269 +5,282 @@ Reasoning with inductive types
 
-     .. note::
-        For destruction of a number, use syntax :n:`destruct (@natural)` (not
-        very interesting anyway).
 
-   .. tacv:: destruct @pattern
+.. _CaseAnalysisAndInduction:
 
-      The argument of :tacn:`destruct` can also be a pattern of which holes are
-      denoted by “_”. In this case, the tactic checks that all subterms
-      matching the pattern in the conclusion and the hypotheses are compatible
-      and performs case analysis using this subterm.
+Case analysis
+-------------
 
-   .. tacv:: destruct {+, @term}
+The tactics in this section implement case
+analysis on inductive or co-inductive objects (see :ref:`variants`).
 
-      This is a shortcut for :n:`destruct @term; ...; destruct @term`.
+.. tacn:: destruct {+, @induction_clause } {? @induction_principle }
 
-   .. tacv:: destruct @term as @or_and_intropattern_loc
+   .. insertprodn induction_clause induction_principle
 
-      This behaves as :n:`destruct @term` but uses the names
-      in :token:`or_and_intropattern_loc` to name the variables introduced in the
-      context. The :token:`or_and_intropattern_loc` must have the
-      form :n:`[p11 ... p1n | ... | pm1 ... pmn ]` with ``m`` being the
-      number of constructors of the type of :token:`term`. Each variable
-      introduced by :tacn:`destruct` in the context of the ``i``-th goal
-      gets its name from the list :n:`pi1 ... pin` in order. If there are not
-      enough names, :tacn:`destruct` invents names for the remaining variables
-      to introduce. More generally, the :n:`pij` can be any introduction
-      pattern (see :tacn:`intros`). This provides a concise notation for
-      chaining destruction of a hypothesis.
+   .. prodn::
+      induction_clause ::= @induction_arg {? as @or_and_intropattern } {? eqn : @naming_intropattern } {? @occurrences }
+      induction_principle ::= using @one_term {? with @bindings } {? @occurrences }
 
-   .. tacv:: destruct @term eqn:@naming_intropattern
-      :name: destruct … eqn:
+   Performs case analysis by generating a subgoal for each constructor of the
+   inductive or co-inductive type in the :n:`@one_term` in :n:`@induction_arg`.
+   In each new subgoal, the tactic replaces :n:`@one_term` with the associated
+   constructor (applied to its arguments).   Unlike :tacn:`induction`,
+   :tacn:`destruct` generates no induction hypothesis.
 
-      This behaves as :n:`destruct @term` but adds an equation
-      between :token:`term` and the value that it takes in each of the
-      possible cases. The name of the equation is specified
-      by :token:`naming_intropattern` (see :tacn:`intros`),
-      in particular ``?`` can be used to let Coq generate a fresh name.
+   The :n:`@one_term` may contain holes that are denoted by “_”. In this case,
+   the tactic checks that all subterms matching the pattern in the conclusion
+   and the hypotheses are compatible and performs case analysis using this subterm.
```

Comment:
I recently used this feature in a proof and this description is wrong: there is no check that all subterms matching the pattern are compatible. The tactic just selects the first subterm matching the pattern that it finds and performs case analysis using this subterm.

Example:
```coq
Require Import Arith.

Goal forall x y, Nat.compare x y = Nat.compare y x.
intros.
destruct (Nat.compare _ _).
```